tasks/determine-image-tag-task.yaml

```diff
@@ -151,13 +157,45 @@ spec:
         return
       }
 
-      # A poor-man's way to differentiate stackrox/stackrox repo from stackrox/collector, stackrox/scanner and others.
+      function determine_tag_suffix() {
+        local -r suffix_param="$(params.TAG_SUFFIX)"
+
+        if [[ "$(is_stackrox_repo)" == "no" ]]; then
+          log "This is not a StackRox repo, tag suffix will be: '${suffix_param}'."
+          echo "${suffix_param}"
+          return
+        fi
+
+        local -r script="scripts/ci/should-konflux-replace-gha-build.sh"
+
+        log "Checking if ${script} thinks the Konflux build should replace the GHA build."
+        local answer
+        answer="$( "${script}" )"
```

Comment:
I'm assuming this script lives in `stackrox/stackrox`. Is it also going to be added to `stackrox/collector`, `stackrox/scanner` and all other places that might use the task in the future? 
All tests in the PR description have run in `stackrox/stackrox`.

Reply:
This script will only be in `stackrox/stackrox`. There's a check `is_stackrox_repo` a few lines above and an early-return if not. I start thinking I should write an ADR explaining the approach for being able to reference it in the future.

Simply: we don't need to apply this trick in `stackrox/collector` and `stackrox/scanner` and in fact e2e tests in collector already expect Konflux-built images with `-fast` suffix. Only in `stackrox/stackrox` the embedded version/tag in Go executables matters at runtime, in `scanner` and `collector` it does not and so we can keep tags/versions with `-fast` suffix there.

Reply:
Ah, I missed this part! Good to know. An ADR or other docs in Confluence regarding: how to tagged/release builds work in Konflux would be great.

Reply:
I'm preparing an ADR here and it will be a part of this same task. https://github.com/stackrox/architecture-decision-records/pull/40
Do you mind approving this PR so I can go ahead and merge it?